File: xper-train/xper-allen/src/org/xper/allen/pga/RegimeType.java
package org.xper.allen.pga;

public enum RegimeType {
    REGIME_ZERO("REGIME_ZERO"),
    REGIME_ONE("REGIME_ONE"),
    REGIME_TWO("REGIME_TWO"),
    REGIME_THREE("REGIME_THREE");

    private final String value;

    RegimeType(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public static RegimeType fromString(String value) {
        for (RegimeType regimeType : RegimeType.values()) {
            if (regimeType.value.equalsIgnoreCase(value)) {
                return regimeType;
            }
        }
        throw new IllegalArgumentException("No enum constant found for value: " + value);
    }
}

File: EStimShapeAnalysis/src/pga/regime_type.py
from enum import Enum


class RegimeType(Enum):
    REGIME_ZERO = "REGIME_ZERO"
    REGIME_ONE = "REGIME_ONE"
    REGIME_TWO = "REGIME_TWO"
    REGIME_THREE = "REGIME_THREE"

    def to_index(self):
        if self == RegimeType.REGIME_ZERO:
            return 0
        elif self == RegimeType.REGIME_ONE:
            return 1
        elif self == RegimeType.REGIME_TWO:
            return 2
        elif self == RegimeType.REGIME_THREE:
            return 3

    @staticmethod
    def from_index(index: int):
        if index == 0:
            return RegimeType.REGIME_ZERO
        elif index == 1:
            return RegimeType.REGIME_ONE
        elif index == 2:
            return RegimeType.REGIME_TWO
        elif index == 3:
            return RegimeType.REGIME_THREE
